### DISCUSSION TITLE
What is your favorite missing editor feature?

### DISCUSSION THREAD

**Discussion by pyscripter** [pyscripter/SynEdit#101] [Polls]
What is your favorite missing editor feature missing from SynEdit?   Your answers may help direct future development efforts.

**Comment by pyscripter** [comment]
Links:
- [Annotations](https://www.scintilla.org/ScintillaDoc.html#Annotations)
- [End-of-line annotations](https://www.scintilla.org/ScintillaDoc.html#EndOfLineAnnotations)

**Reply by MShark67** [reply]
I voted for variable size fonts, but as mentioned previously, it's not something I'd consider a mainstream need for a coding editor.  I had some clients request this and so I've added the ability to a forked version of SynEdit.  I needed to remove a few things to work properly.  Specifically, it had to not allow bold or italic characters as code highlighting options (I just ignore them in the paint code.)  With those removed it actually works very well with variable sized fonts (and falls back to showing bold/italic if the selected font is monospaced.)  I'm happy to share the code if it would help in any way, but I suspect that if you did it it would be less "hacky" than mine lol.

**Reply by MShark67** [reply]
if we could multi-vote I'd vote for the annotated scrollbars or the minimap.

**Reply by pyscripter** [reply]
Unfortunately Github polls do not allow that.

**Reply by pyscripter** [reply]
The next couple of things I will work on is annotated scrollbars and zoom commands.
If you have some time, you could work on the minimap implementation.  The VS-code features could serve as a target.  
- A simplistic approach can be found [here](https://github.com/SynEdit/SynEdit/tree/master/Demos/MiniMap).
- [CudaText](https://github.com/Alexey-T/CudaText)  has a more advanced [implementation](https://github.com/Alexey-T/ATSynEdit/blob/master/atsynedit/atsynedit.pas#L4932).  It also has something called micromap that looks interesting

Also please let me know when you feel the multicaret stuff is solid enough to be pushed to master.

**Comment by DGH2112** [comment]
I would multi-vote as follows (in no order of priority):

1. Annotated Scrollbars as in Delphi 12.2 and VS code
2. Annotations
3. End-of-line annotations

**Comment by puppadrillo** [comment]
Probably it is already there but CTRL+F will search a next occurrence of the current selection and SHIFT+CTRL+F search the previous one.  Useful quick commands. Also a SHIFT+CLICK will swap true to false and vice versa.

**Comment by pyscripter** [comment]
#102 (Zoom In/Out/Reset  commands) implemented in multicaret branch.

**Comment by pyscripter** [comment]
#103 (Annotated Scrollbars) have been implemented in the multicaret branch.
See this [post](https://en.delphipraxis.net/topic/12417-synedit-now-has-annotated-scrollbars).

**Comment by pyscripter** [comment]
Structural highlighting has been implemented in the multicaret branch:

![image](https://github.com/user-attachments/assets/738041f4-c72f-4a84-9602-b96af8d330fe)


See #104 for details.

Flow control visualization to follow.

**Comment by pyscripter** [comment]
Flow control symbols have also been added:

![image](https://github.com/user-attachments/assets/0d72547a-d33d-4421-b96f-f2e760df42bf)

See #105 for details

**Comment by Philnext** [comment]
Not in the list, but **SynEdit for FMX** is my prefered feature.
